src/main/java/com/rylinaux/plugman/util/BukkitCommandWrap.java

```diff
@@ -177,92 +125,193 @@ public void sync() {
 
     public void unwrap(String command) {
         if (this.nmsVersion == null) return;
-        if (this.minecraftServerClass == null) try {
-            this.minecraftServerClass = Class.forName("net.minecraft.server." + this.nmsVersion + ".MinecraftServer");
-        } catch (ClassNotFoundException e) {
-            try {
-                this.minecraftServerClass = Class.forName("net.minecraft.server.MinecraftServer");
-            } catch (ClassNotFoundException classNotFoundException) {
-                classNotFoundException.printStackTrace();
-                classNotFoundException.addSuppressed(e);
-                return;
-            }
-        }
-        if (this.getServerMethod == null) try {
-            this.getServerMethod = this.minecraftServerClass.getMethod("getServer");
-            this.getServerMethod.setAccessible(true);
-        } catch (NoSuchMethodException e) {
-            e.printStackTrace();
-            return;
-        }
 
-        Object server;
+        if (!resolveMcServerClass()) return;
+
+        if (!resolveGetServerMethod()) return;
+        Object server = getServerInstance();
+
+        if (!resolveVanillaCommandDispatcherField()) return;
+        Object commandDispatcher = getCommandDispatcher(server);
+
+        if (!resolveBField()) return;
+
+        CommandDispatcher b = getDispatcher(commandDispatcher);
+        if (b == null) return;
+
+        if (resolveRemoveCommandMethod()) return;
 
         try {
-            server = this.getServerMethod.invoke(this.minecraftServerClass);
+            this.removeCommandMethod.invoke(b.getRoot(), command);
         } catch (IllegalAccessException | InvocationTargetException e) {
             e.printStackTrace();
-            return;
         }
+    }
+
+    private boolean resolveRemoveCommandMethod() {
+        if (this.removeCommandMethod == null) try {
+            try {
+                this.removeCommandMethod = RootCommandNode.class.getDeclaredMethod("removeCommand", String.class);
+            } catch (NoSuchMethodException | NoSuchMethodError ex) {
+                this.removeCommandMethod = CommandNode.class.getDeclaredMethod("removeCommand", String.class);
+            }
+        } catch (NoSuchMethodException e) {
+            e.printStackTrace();
+            return true;
+        }
+        return false;
+    }
 
-        if (this.vanillaCommandDispatcherField == null) try {
-            this.vanillaCommandDispatcherField = this.minecraftServerClass.getDeclaredField("vanillaCommandDispatcher");
-            this.vanillaCommandDispatcherField.setAccessible(true);
-        } catch (NoSuchFieldException e) {
+    private @Nullable CommandDispatcher getDispatcher(Object commandDispatcher) {
+        CommandDispatcher b = null;
+        try {
+            b = (CommandDispatcher) this.bField.get(commandDispatcher);
+        } catch (IllegalAccessException e) {
             e.printStackTrace();
-            return;
         }
+        return b;
```

Comment:
Please move this to line #168 and return null instead.

You should be able to remove line #166 by doing so

Reply:
Same thing goes for the other methods, of course

Reply:
Updating...